packages/next/client/index.js

```diff
@@ -173,8 +173,17 @@ export default async ({ webpackHMR: passedWebpackHMR } = {}) => {
       render({ App, Component, props, err, emitter })
     }
   })
-
-  render({ App, Component, props, err: initialErr, emitter })
+  const renderCtx = { App, Component, props, err: initialErr, emitter }
+  render(renderCtx)
+
+  if (window.__NEXT_DATA__.skeleton && Component.getInitialProps) {
+    props.pageProps = await Component.getInitialProps({
```

Comment:
This is missing `AppTree: wrapApp(App)`, right? This might be pending on the other PR to add this here, but might be worth adding now.

Also, does the App GIP need invoked? Seems like an omission to me.

Reply:
We haven't [merged](https://github.com/zeit/next.js/pull/8223) exposing `AppTree` in `NextPageContext` yet. We should probably call `App` `getInitialProps` for consistency so updated that